lib/jekyll/commands/serve.rb

```diff
@@ -1,142 +1,192 @@
-# -*- encoding: utf-8 -*-
 module Jekyll
   module Commands
     class Serve < Command
-
       class << self
+        COMMAND_OPTIONS = {
+          "detact" => ["-B", "--detach", "Run the server in the background (detach)"],
+          "port" => ["-P", "--port [PORT]", "Port to listen on"],
+          "host" => ["host", "-H", "--host [HOST]", "Host to bind to"],
+          "skip_initial_build" => ["skip_initial_build", "--skip-initial-build", "Skips the initial site build" \
+            "which occurs before the server is started."],
+
+          "enable_ssl" => ["--enable-ssl", "Enable WEBrick SSL."],
+          "ssl_cert" => ["--ssl-cert [CERT]", "X.509 (SSL) certificate."],
+          "ssl_key" => ["--ssl-key [KEY]", "X.509 (SSL) Private Key."],
+          "baseurl" => ["-b", "--baseurl [URL]", "Base URL"]
+        }
 
         def init_with_program(prog)
           prog.command(:serve) do |c|
-            c.syntax 'serve [options]'
-            c.description 'Serve your site locally'
+            c.description "Serve your site locally"
+            c.syntax "serve [options]"
             c.alias :server
             c.alias :s
 
             add_build_options(c)
+            COMMAND_OPTIONS.each do |k, v|
+              c.option k, *v
+            end
+
+            c.action do |a, o|
```

Comment:
`a` and `o` are cryptic – would you mind changing them back to abbreviated words? I don't think `args` is used, so you can `_` that, and call `o` `opts`.